eng/scripts/Attest-CPEX-KPIs.ps1

```diff
@@ -0,0 +1,127 @@
+Set-StrictMode -Version 3
+. (Join-Path $PSScriptRoot "\..\common\scripts\Helpers\DevOps-WorkItem-Helpers.ps1")
+
+$Incomplete = 0
+$Complete = 1
+$NA = 3
+
+# Add KPI Ids after launch criteria creation
+$KPI_ID_Onboarding_Private_Preview = "Onboarding_Private_Preview"
+$KPI_ID_Mgmt_Private_Preview = "Mgmt_Private_Preview"
+$KPI_ID_Data_Private_Preview = "Data_Private_Preview"
+$KPI_ID_Mgmt_Public_Preview = "Mgmt_Public_Preview"
+$KPI_ID_Data_Public_Preview = "Data_Public_Preview"
+$KPI_ID_Mgmt_GA = "Mgmt_GA"
+$KPI_ID_Data_GA = "Data_GA"
+
+function InvokeKustoCommand($command) {
+    $clusterUri = "https://azsdk-cpex-attestation.westus2.kusto.windows.net"
+    $databaseName = "CPEX-attestations"
+    $accessToken = az account get-access-token --resource "https://api.kusto.windows.net" --query "accessToken" --output tsv
+    $headers = @{ Authorization="Bearer $accessToken" }
+    Invoke-RestMethod -Uri "$clusterUri/v1/rest/mgmt" -Headers $headers -Method Post -Body (@{csl=$command; db=$databaseName} | ConvertTo-Json -Depth 3) -ContentType "application/json"
+}
+
+function AddAttestationEntry($targetId, $actionItemId, $status, $targetType, $url) {
+    $tableName = "KPI_Attestations"
+
+    $command = @"
+.append $tableName <|
+print
+    TargetId = "$targetId",
+    ActionItemId = "$actionItemId",
+    Status = int($status),
+    TargetType = "$targetType",
+    CreatedTime = datetime(now),
+    EvidenceUrl = "$url"
+"@
+
+    InvokeKustoCommand $command
+}
+
+$triages =  Get-TriagesForCPEXAttestation
+
+foreach ($triage in $triages) {
+    $fields = $triage.fields
+    $dataScope = $fields["Custom.DataScope"]
+    $mgmtScope = $fields["Custom.MgmtScope"]
+    $productServiceTreeId = $fields["Custom.ProductServiceTreeID"]
+    $productType = $fields["Custom.ProductType"]
+    $url = $triage.url
+
+    AddAttestationEntry $productServiceTreeId $KPI_ID_Onboarding_Private_Preview $Complete $productType $url
```

Comment:
does this mean that for onboarding we really don't need to know if it was management plane or data plane attestation, right?

Reply:
Not sure I understand why when attesting onboarding, the other states are stablished. Is this a requirement?
What if by race condition a release plan was completed, the attestation done, but then onboarding kicked in and the values previously added for the release planner get overwritten by this?